portfolio/src/main/java/com/google/sps/servlets/CommentsController.java

```diff
@@ -14,36 +14,239 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.models.Info;
 import com.google.sps.models.Comment;
+import com.google.sps.models.Comments;
 import java.io.IOException;
-import java.util.List;
 import java.util.ArrayList;
-import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
 
 @WebServlet("/comments")
 public class CommentsController extends HttpServlet {
-  private List<Comment> comments;
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
-  public void init() {
-    comments = new ArrayList<>();
-    for(Integer i = 0; i < 5; i++){
-      comments.add(0, new Comment("Foo", "Bar", new Date()));
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    // Parse parameters
+    int page = parsePageParam(request.getParameter("page"));
+    int pageSize = parsePageSizeParam(request.getParameter("pageSize"));
+    SortDirection order = parseOrderParam(request.getParameter("order"));
+    String orderBy = parseOrderByParam(request.getParameter("orderBy"));
+
+    Query query = new Query("Comment").addSort(orderBy, order);
+    PreparedQuery results = datastore.prepare(query);
```

Comment:
The way pagination works is that it retrieves some Info for the comments query, and receives some parameters, which I'm listing:

**Parameters**
- `page`: Int
- `pageSize`: Int
- `order`: enum: ['asc', 'desc']
- `orderBy`: enum: ['createdAt', 'likes']

**Info**
- `count`: This is the total of entities
- `pages`: The total of pages that can be built using the requested parameters
- `prev`: If exists, returns the number of the previous page
- `next`: If exists, returns the number of the next page

This structure needs some features not available for data store, such as counting entities, skips, etc.
I was searching for some info about this but I only found the following:

[Cursors](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Cursor) and [FetchOptions](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions)

In the case of Cursors, they only save a position to resume a query, which requires the user to make the query at first, and obviously this is not particularly friendly with `pageSize` parameter.

For FetchOptions, it allows me to set a `limit` and and `offset`, and I'm listing their caveats:
- Limit is exactly what pageSize needs, but since there is no way to count the total documents, this option disallows me to send total count parameter, and also, to calculate the total of pages.
- Offset is exactly what page needs, but according to docs, it doesn't just skip the entities, it also traverses the whole datastore until the offset is met, which is not the most performant option

The way it was implemented works really well, but I know that is not as efficient as I would like. I'm just detailing this to see if you guys have some thoughts on this

Reply:
No problem, I just wanted to be sure you or Carlos knew what was going up here.

I added the corresponding comments you are suggesting in #12, so I'm closing and merging this PR